test/e2e/ComplexShrink.spec.ts

```diff
@@ -0,0 +1,92 @@
+import * as fc from '../../src/fast-check';
+import { seed } from './seed';
+
+describe(`ComplexShrink (seed: ${seed})`, () => {
+  it('Should shrink two integers linked by a non-symmetric relation', () => {
```

Comment:
I tried to reproduce this test with Hypothesis (Python framework doing PBT). The Python snippet is:

```py
from hypothesis import given
from hypothesis.strategies import integers

@given(a=integers(0, 10000), b=integers(0, 10000))
def non_sym(a, b):
    def status():
        if a < 1000:
            return True
        if b < 1000:
            return True
        if b < a:
            return True
        if abs(a - b) < 10:
            return True
        return b - a >= 1000
    assert status()

if __name__ == "__main__":
    non_sym()
```

And the shrinker perfectly reaches the expected solution by relying on the following shrinking path:

```txt
0, 0, True
5307, 0, True
5307, 1120, True
0, 0, True
5282, 0, True
5282, 2870, True
1489, 0, True
1489, 2366, False
  2366, 2366, True
  1489, 1489, True
  2366, 2366, True
  6510, 8457, True
  8457, 6510, True
  6510, 6510, True
  6424, 6134, True
  6424, 6424, True
  1489, 2366, False
    1489, 2366, False
      1489, 1, True
      1489, 318, True
      1489, 147, True
      1489, 62, True
      1489, 1183, True
      1489, 2364, False
        1489, 2362, False     -2
          1489, 2360, False     -2
            1489, 2358, False     -2
              1489, 2356, False     -2
                1489, 2346, False     -10
                  1489, 2326, False     -20
                    1489, 2286, False     -40
                      1489, 2206, False     -80
                        1489, 2046, False     -160
                          1489, 1726, False     -320
                            1489, 1086, True    -640
                            1489, 1406, True    -320
                            1489, 1566, False   -160
                              1489, 1486, True    -80
                              1489, 1526, False   -40
                                1489, 1506, False   -20
                                  1489, 1496, True    -10
                                  1489, 1502, False   -4
                                    1489, 1500, False   -2
                                      1489, 1498, True    -2
                                      1489, 1499, False   -1
                                        1489, 475, True
                                        1489, 187, True
                                        1489, 219, True
                                        1489, 749, True
                                        1489, 1497, True
                                        0, 1499, True
                                        1, 1499, True
                                        465, 1499, True
                                        186, 1499, True
                                        209, 1499, True
                                        744, 1499, True
                                        1487, 1499, False
                                          0, 12, True
                                          1, 13, True
                                          463, 475, True
                                          185, 197, True
                                          207, 219, True
                                          743, 755, True
                                          1485, 1497, False
                                            1483, 1495, False
                                              1481, 1493, False
                                                1479, 1491, False
                                                  1477, 1489, False
                                                    1467, 1479, False
                                                      1447, 1459, False
                                                        1407, 1419, False
                                                          1327, 1339, False
                                                            1167, 1179, False
                                                              847, 859, True
                                                              1007, 1019, False
                                                                927, 939, True
                                                                967, 979, True
                                                                987, 999, True
                                                                997, 1009, True
                                                                1003, 1015, False
                                                                  1001, 1013, False
                                                                    999, 1011, True
                                                                    1000, 1012, False
                                                                      1485, 1012, True
                                                                      1486, 1012, True
                                                                      0, 1012, True
                                                                      1, 1012, True
                                                                      488, 1012, True
                                                                      250, 1012, True
                                                                      232, 1012, True
                                                                      500, 1012, True
                                                                      998, 1012, True
                                                                      999, 1012, True
                                                                      1000, 0, True
                                                                      1000, 1, True
                                                                      1000, 500, True
                                                                      1000, 253, True
                                                                      1000, 244, True
                                                                      1000, 506, True
                                                                      1000, 1010, False
                                                                        1000, 1008, True
                                                                        1000, 1009, True
                                                                        1000, 498, True
                                                                        1000, 252, True
                                                                        1000, 242, True
                                                                        1000, 505, True
                                                                        0, 1010, True
                                                                        1, 1010, True
                                                                        488, 1010, True
                                                                        250, 1010, True
                                                                        232, 1010, True
                                                                        500, 1010, True
                                                                        998, 1010, True
                                                                        999, 1010, True
                                                                        999, 1009, True
                                                                        0, 5, True
                                                                        1000, 769, True
                                                                        769, 1010, True
                                                                        1000, 777, True
                                                                        777, 1010, True
                                                                        1000, 1010, False
```